### ISSUE TITLE
Filter issue after 3.0.2.0 to 3.0.3.2 upgrade.

### ISSUE THREAD

**Issue by escservices** [opencart/opencart#7509]
**What version of OpenCart are you reporting this for?**
3.0.3.2
**Describe the bug**
Product filters, when combining filters it used to show products that match both and exclude any that don't meet both criteria.  After updating from 3.0.2.0 to 3.0.3.2 this seems to now be an OR filter instead of an AND filter which it was in 3.0.20.

**To Reproduce**
Select filters on product category page and products shown are ones that match Filter A or Filter B.

**Expected behavior**
When selecting 2 filters it should only show products that match both.  For instance if I pick shoe size 11 and color black I shouldn't receive a list of all different color size 11 shoes.  This used to work fine prior to the 3.0.2.0 to 3.0.3.2 upgrade. 

**Server / Test environment (please complete the following information):**
 - Hosted Web Server
 - Centos 7 Core
 - PHP Version 7.3 (Tested on 5.6-7.3)
 - Apache version 2.4? (Managed by web host)
 - Browser(s) tested with [chrome]
 - Version [75.0.3770.100 (Official Build) (64-bit)]

**Additional context**

**Comment by straightlight** [comment]
Any URL to show this reported incident?

**Comment by escservices** [comment]
You could look at the link below but the store was built for a very specific industry so the filters may not make sense to the anyone who doesn't know anything about collectible currency.  Basically the filters currently selected are $1 and PMG, notice the items it lists are both PCGS and PMG (you can tell based on the logo in the top left corner of each product image).  If you remove the 1$ filter it will display only PMG but once you add the $1 dollar filter it basically ignores the other filters.  It seems to be random on which filter it decides to use after you've selected more than 1.  This happens on every product page that has filters across the whole site.  

https://centurialcollectibles.com/Small-Size/Silver-Certificates&filter=4,9

**Comment by straightlight** [comment]
You are using an SEO extension. Without the SEO URL activated, can this issue be reproduced?

**Comment by ADDCreative** [comment]
From memory it's always been the case the filter used OR. To get a much more useful AND filter you need to make modifications or use an extension. https://github.com/opencart/opencart/issues/636

Also there have been no changes between 3.0.2.0 and 3.0.3.2 that would directly affect the filters.

Is it possible you had a custom modification that was overwritten by the upgrade to 3.0.3.2 or an extension that is now no longer working?

**Comment by escservices** [comment]
I figured it out while writing my reply below, I did recall having this issue a long time ago and it was a VQmod that I had installed that changed the filter functionality.  I think it confused me because VQMODS don't show up in that modifications section of the opencart dashboard.  To be honest I forgot VQmods used to be a thing, I think they're deprecated now and I guess that last updated broke their functionality?  Either way I found a free OCMOD that worked great I put the link below.  Thanks for taking the time to try to help me, hopefully someone else finds this info useful as well.  

https://www.opencart.com/index.php?route=marketplace/extension/info&member_token=PNcKmnCPRrNjKH0fjqhg2gBtY5mWEDde&extension_id=23603&filter_license=0

![image](https://user-images.githubusercontent.com/1287511/61509255-adb7e780-a9bb-11e9-822a-559f9d0343ec.png)


Regarding the SEO, I'm using the built in one and disabling it didn't make a difference but good idea!  

Regarding the custom modification we didn't have anything besides a modification that shows a sold out logon on sold out items.  

We've been using opencart since like 2.0.1.0 I believe and I've done a few upgrades and never seen the filter change functionality.  Our customer that pointed it out to us uses the site weekly for over a year and just recently said it was an issue.  The only changes I made were upgrading to 3.0.3.2 and upgrading the PHP version.  I did try lowering the PHP version to see if that was it but no luck.
